DEMS-Common/src/UDP/Message.java

```diff
@@ -46,9 +46,13 @@ of this software and associated documentation files (the "Software"), to deal
     private InetAddress m_Addr;
     private int m_Port;
 
+    // @warning max data.length is 2048 bytes !
     // Main entry point for initiating communication, See Socket for an example
-
-    public Message(OperationCode code, int seq, String data, AddressBook addrInfo) {
+    public Message(OperationCode code, int seq, String data, AddressBook addrInfo) throws Exception {
+        if(data.length() >= 2048){
```

Comment:
Out of curiosity, why 2048 bytes?

Reply:
making a byte stream of a complicated object can get pretty large, didnt want a hard coded limit being too small